### PR TITLE
CAMEL-16577 add support to set the index time

### PR DESCRIPTION
**Pull request by iridian-ks**
Add new option in the splunk-hec component in order to set the time the event occurred rather than defaulting the time to the index time.

### CODE REVIEW

**Comment by davsclaus** [comment]
I wonder if you wont want to have a header with the time that can take precedence, instead of endpoint option. Endpoint options tend to be more static.

**Review by oscerd — CHANGES_REQUESTED** [review]
Yeah, it makes sense to have a time header too with precedence.

**Comment by iridian-ks** [comment]
Thanks for the feedback!!!

To be clear. What is the desired approach here?

1. Header ONLY
2. Header AND option and Header takes precedence
3. Option AND header and Option takes precedence

My initial idea was to use `.toD("${header.splunkTime}"). So, yes the header is pretty much always required when using this feature. Thank you for the time and guidance in this! :)

**Comment by davsclaus** [comment]
Option2 is what other components would do.

**Comment by davsclaus** [comment]
Do you have time to work on this, we are getting close to be ready for building the next release.

**Comment by iridian-ks** [comment]
Thanks for the reminder @davsclaus 

I pushed a new commit that adds support for the header as well as some update to the documentation. I am definitely still learning the ropes of best practices so feedback is welcomed!

Thanks for your time on this.

File: components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java [hunk 1]
```diff
@@ -128,5 +131,25 @@ private void buildPayload(Map<String, Object> payload) {
         if (endpoint.getConfiguration().getHost() != null) {
             payload.put("host", endpoint.getConfiguration().getHost());
         }
+
+        Object timeInput = message.getHeader(SplunkHECComponent.INDEX_TIME, endpoint.getConfiguration().getTime());
```
Comment (davsclaus) [inline components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java:135]: You should use Camel's type converter, and use getHeader where you can specify the returned time as Long so Camel deal with type conversions. Then all the code below can be removed

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java (hunk 1 above)
Comment (iridian-ks) [reply components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java]: Done

**Review by davsclaus — APPROVED** [review]

**Pull request merged**